pkg/lintcontext/scheme.go

```diff
@@ -0,0 +1,13 @@
+package lintcontext
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+// UseCustomScheme allows a user to set a scheme that is used for deserializing the k8s objects
+// This can be used to add support for deserializing non-native k8s objects that implement the v1.Object interfeace (e.g. custom resources)
+func UseCustomScheme(s *runtime.Scheme) {
```

Comment:
Hmm, instead of this, I would prefer to make this an argument to `CreateContexts` as you had suggested in your issue. I am hesitant to expose a mechanism for mutating global variables, since that will mean that different parts of a program can have unexpected side-effects on each other.

My suggestion would be as follows:
1. Add a new function called `CreateContextsWithOptions`, that takes in a first parameter which is a struct of options. Currently, the only member of the struct will be a `customDecoder`. This `customDecoder` can be passed through into the `lintContextImpl`, all the way into `parseObjects` (and if the `customDecoder` is `nil`, that falls back to the original decoder).
1. Change the implementation of `CreateContexts` to just call `CreateContextsWithOptions` with an empty `Options{}` struct to avoid code duplication.

This will a) ensure no breakage of backward compatibility for any clients using `CreateContexts` and b) ensure that any further options we may want to add to `CreateContextsWithOptions` require no function signature change, only the addition of fields to the `Options` struct.

Does this make sense?

Reply:
yup! Updated PR accordingly